ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala

```diff
@@ -90,13 +96,20 @@ private[index] class IndexServiceImpl(
       to.foreach(offset =>
         Spans.setCurrentSpanAttribute(SpanAttribute.OffsetTo, offset.toHexString)
       )
-      dispatcher
+      val flatTransactionsSource = dispatcher
         .startingAt(
           from.getOrElse(Offset.beforeBegin),
           RangeSource(transactionsReader.getFlatTransactions(_, _, convertFilter(filter), verbose)),
           to,
         )
         .map(_._2)
+
+      InstrumentedSource
```

Comment:
Can you help me understand what this instrumentation is good for for client streams? I am thinking: instrumenting this buffer helps us understand whether a client stream is faster or slower than the producer, but we cannot really map this back to an actual client stream, or can we? What does this help us to understand better?

Reply:
You are right. We had multiple talks also with @mziolekda over this. Ideally we should be able to tag each metric in order to individualize the stream performance measurement for each consumer, but that is not currently possible (afair we have a task/epic for improving our metrics tech stack that could allow this). Until then, this is a good enough proxy for some of the applications. @mziolekda what do you think?

Reply:
correct, it's not ideal but better than nada